java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java

```diff
@@ -70,154 +71,103 @@ public class AwsSdk2Transport implements OpenSearchTransport {
      * @param options Options that apply to all requests. Can be null. Create with
      *                {@link AwsSdk2TransportOptions#builder()} and use these to specify non-default credentials,
      *                compression options, etc.
      */
     public AwsSdk2Transport(
-            @Nonnull SdkHttpClient httpClient,
+            @CheckForNull SdkAsyncHttpClient asyncHttpClient,
             @Nonnull String host,
             @Nonnull Region signingRegion,
             @CheckForNull AwsSdk2TransportOptions options) {
-        this(httpClient, null, host, "es", signingRegion, options);
+        this(asyncHttpClient, host, "es", signingRegion, options);
     }
 
     /**
      * Create an {@link OpenSearchTransport} with a synchronous AWS HTTP client.
-     * <p>
-     * Note that asynchronous OpenSearch requests sent through this transport will be dispatched
-     * *synchronously* on the calling thread.
      *
-     * @param httpClient HTTP client to use for OpenSearch requests.
+     * @param httpClient Synchronous HTTP client to use for OpenSearch requests.
      * @param host The fully qualified domain name to connect to.
-     * @param signingServiceName The AWS signing service name, one of `es` (Amazon OpenSearch) or `aoss` (Amazon OpenSearch Serverless).
      * @param signingRegion The AWS region for which requests will be signed. This should typically match the region in `host`.
      * @param options Options that apply to all requests. Can be null. Create with
      *                {@link AwsSdk2TransportOptions#builder()} and use these to specify non-default credentials,
      *                compression options, etc.
      */
     public AwsSdk2Transport(
-            @Nonnull SdkHttpClient httpClient,
-            @Nonnull String host,
-            @Nonnull String signingServiceName,
-            @Nonnull Region signingRegion,
-            @CheckForNull AwsSdk2TransportOptions options) {
-        this(httpClient, null, host, signingServiceName, signingRegion, options);
-    }
-
-    /**
-     * Create an {@link OpenSearchTransport} with an asynchronous AWS HTTP client
-     * <p>
-     * Note that synchronous OpenSearch requests sent through this transport will be dispatched
-     * using the asynchronous client, but the calling thread will block until they are complete.
-     *
-     * @param asyncHttpClient HTTP client to use for OpenSearch requests.
-     * @param host The target host.
-     * @param signingRegion The AWS region for which requests will be signed. This should typically match region in `host`.
-     * @param options Options that apply to all requests. Can be null. Create with
-     *                {@link AwsSdk2TransportOptions#builder()} and use these to specify non-default credentials,
-     *                compression options, etc.
-     */
-    public AwsSdk2Transport(
-            @Nonnull SdkAsyncHttpClient asyncHttpClient,
+            @CheckForNull SdkHttpClient httpClient,
```

Comment:
Just a minor comment: Should we call this `syncHttpClient`?

Reply:
Good catch, fixed.